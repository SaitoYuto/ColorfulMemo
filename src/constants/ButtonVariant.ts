export const BUTTON_VARIANT = {
  TEXT: "text",
  TONAL: "tonal",
  FLAT: "flat",
  ELEVATED: "elevated",
  OUTLINED: "outlined",
  PLAIN: "plain",
  UNDEFINED: undefined,
} as const;
