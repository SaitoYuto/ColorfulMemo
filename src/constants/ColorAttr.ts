export const COLOR_ATTR = {
  DEFAULT: {
    ALERT: "#757575",
    BAR: "#9E9E9E",
    BUTTON: "#9E9E9E",
    DIVIDER: "#757575",
    ICON: "#9E9E9E",
    LIST: "#757575",
    TEXT_FIELD: "#F5F5F5",
    TEXT_AREA: "#F5F5F5",
    SELECT: "#F5F5F5",
  } as const,
  BLUE: {
    ALERT: "rgb(33, 150, 243)",
    BAR: "#2196F3",
    BUTTON: "#2196F3",
    DIVIDER: "rgb(33, 150, 243)",
    ICON: "#2196F3",
    LIST: "rgb(33, 150, 243)",
    TEXT_FIELD: "#BBDEFB",
    TEXT_AREA: "#BBDEFB",
    SELECT: "#BBDEFB",
  } as const,
  RED: {
    ALERT: "rgb(176, 0, 32)",
    BAR: "#F44336",
    BUTTON: "#F44336",
    DIVIDER: "rgb(176, 0, 32)",
    ICON: "#F44336",
    LIST: "rgb(176, 0, 32)",
    TEXT_FIELD: "#FFCDD2",
    TEXT_AREA: "#FFCDD2",
    SELECT: "#FFCDD2",
  } as const,
  YELLOW: {
    ALERT: "rgb(251, 140, 0)",
    BAR: "#FFEB3B",
    BUTTON: "#FFEB3B",
    DIVIDER: "rgb(251, 140, 0)",
    ICON: "#FFEB3B",
    LIST: "rgb(251, 140, 0)",
    TEXT_FIELD: "#FFF9C4",
    TEXT_AREA: "#FFF9C4",
    SELECT: "#FFF9C4",
  } as const,
  GREEN: {
    ALERT: "rgb(76, 175, 80)",
    BAR: "#4CAF50",
    BUTTON: "#4CAF50",
    DIVIDER: "rgb(76, 175, 80)",
    ICON: "#4CAF50",
    LIST: "rgb(76, 175, 80)",
    TEXT_FIELD: "#C8E6C9",
    TEXT_AREA: "#C8E6C9",
    SELECT: "#C8E6C9",
  } as const,
} as const;
