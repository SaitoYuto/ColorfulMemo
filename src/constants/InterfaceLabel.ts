export const INTERFACE_LABEL = {
  ALERT: {
    NO_MEMO: "No memos",
  } as const,
  BUTTON: {
    CANCEL: "Cancel",
    DELETE: "Delete",
    SAVE: "Save",
    UPDATE: "Update",
  } as const,
  COMMON: {
    TITLE: "Colorful Memo",
  } as const,
  INPUT: {
    MEMO_CONTENT: "*Content",
    MEMO_TITLE: "*Title",
  } as const,
  PAGES: {
    ADD: "New",
    EDIT: "Edit",
    HOME: "Home",
    SETTING: "Setting",
  } as const,
  SELECT: {
    COLOR: "Theme Color",
    MODE: "Theme Mode",
    BUTTON: "Button Variant",
  } as const,
  TIME_FRAME: {
    TODAY: "Today",
    LAST_WEEK: "Last Week",
    LAST_MONTH: "Last Month",
    MORE_THAN_MONTH: "More than 30 days",
  } as const,
} as const;
