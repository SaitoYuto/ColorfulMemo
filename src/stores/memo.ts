import { defineStore } from "pinia";
import type { Memo } from "@/types/Memo";

export const memo = defineStore("memo", {
  state: () => {
    return {
      memos: [] as Memo[],
    };
  },
  getters: {
    getMemos: (state): Memo[] => {
      return state.memos;
    },
    length: (state): number => {
      return state.memos.length;
    },
  },
  actions: {
    add(memo: Memo): void {
      this.memos.unshift(memo);
    },
    findById(id: string): Memo | undefined {
      return this.memos.find((memo) => memo.id === id);
    },
    updateById(id: string, title: string, content: string): void {
      const target = this.memos.find((memo) => memo.id === id);
      if (!target) {
        return;
      }
      target.title = title;
      target.content = content;
      target.dateTime =
        new Date().toLocaleDateString() + " " + new Date().toLocaleTimeString();
    },
    removeById(id: string): void {
      this.memos = this.memos.filter((memo) => memo.id !== id);
    },
  },
  persist: {
    storage: localStorage,
  },
});
