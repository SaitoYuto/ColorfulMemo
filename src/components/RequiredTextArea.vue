<template>
  <v-textarea
    :bg-color="bgColor"
    :class="$props.class"
    :clearable="clearable"
    :label="label"
    :rules="[isRequired]"
    :variant="variant"
    v-model="input"
  ></v-textarea>
</template>

<script lang="ts" setup>
import { MESSAGE } from "@/constants/Message";
import { TEXTAREA_VARIANT } from "@/constants/TextAreaVariant";
import type { TextAreaVariant } from "@/types/TextAreaVariant";

defineProps({
  bgColor: String,
  class: {
    type: String,
    default: "mt-4",
  },
  clearable: {
    type: Boolean,
    default: true,
  },
  label: String,
  variant: {
    type: String as PropType<TextAreaVariant>,
    default: TEXTAREA_VARIANT.OUTLINED,
  },
});

const input = defineModel("input", {
  type: String,
  default: "",
});

/**
 * check if the input exists
 *
 * @returns {boolean | string}
 */
function isRequired(): string | boolean {
  return input.value.length > 0 || MESSAGE.FIELD.REQUIRED;
}
</script>
