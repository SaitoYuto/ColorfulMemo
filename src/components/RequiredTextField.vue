<template>
  <v-text-field
    :bg-color="bgColor"
    :class="$props.class"
    :clearable="clearable"
    :label="label"
    :rows="rows"
    :rules="[isRequired]"
    :variant="variant"
    v-model="input"
  ></v-text-field>
</template>

<script lang="ts" setup>
import { MESSAGE } from "@/constants/Message";
import { TEXTFIELD_VARIANT } from "@/constants/TextFieldVariant";
import type { TextFieldVariant } from "@/types/TextFieldVariant";

defineProps({
  bgColor: String,
  clearable: {
    type: Boolean,
    default: true,
  },
  label: String,
  variant: {
    type: String as PropType<TextFieldVariant>,
    default: TEXTFIELD_VARIANT.OUTLINED,
  },
  rows: {
    type: String,
    default: "1",
  },
  class: {
    type: String,
    default: "mt-4",
  },
});

const input = defineModel("input", {
  type: String,
  default: "",
});

/**
 * check if the input exists
 *
 * @returns {boolean | string}
 */
function isRequired(): boolean | string {
  return input.value.length > 0 || MESSAGE.FIELD.REQUIRED;
}
</script>
