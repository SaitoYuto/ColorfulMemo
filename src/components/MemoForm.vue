<template>
  <v-form ref="memoForm">
    <v-container>
      <RequiredTextField
        data-testid="title-text-field"
        :bgColor="colorAttr.TEXT_FIELD"
        :label="INTERFACE_LABEL.INPUT.MEMO_TITLE"
        v-model:input="titleInput"
      />
      <RequiredTextArea
        data-testid="content-text-area"
        :bgColor="colorAttr.TEXT_AREA"
        :label="INTERFACE_LABEL.INPUT.MEMO_CONTENT"
        v-model:input="contentInput"
      />
    </v-container>
    <v-container class="d-flex justify-space-evenly mx-16">
      <BaseButton
        data-testid="save-button"
        :color="colorAttr.BUTTON"
        :text="INTERFACE_LABEL.BUTTON.SAVE"
        :variant="buttonVariant"
        :isShow="!isEdit()"
        @click="save"
      />
      <BaseButton
        data-testid="update-button"
        :color="colorAttr.BUTTON"
        :text="INTERFACE_LABEL.BUTTON.UPDATE"
        :variant="buttonVariant"
        :isShow="isEdit()"
        @click="update"
      />
      <BaseButton
        data-testid="delete-button"
        :color="colorAttr.BUTTON"
        :text="INTERFACE_LABEL.BUTTON.DELETE"
        :variant="buttonVariant"
        :isShow="isEdit()"
        @click="remove"
      />
      <BaseButton
        data-testid="cancel-button"
        :color="colorAttr.BUTTON"
        :text="INTERFACE_LABEL.BUTTON.CANCEL"
        :variant="buttonVariant"
        @click="forwardHome"
      />
    </v-container>
  </v-form>
</template>

<script lang="ts" setup>
import moment from "moment";
import { INTERFACE_LABEL } from "@/constants/InterfaceLabel";
import { memo } from "@/stores/memo";
import type { Memo } from "@/types/Memo";
import { ButtonVariant } from "@/types/ButtonVariant";

const props = defineProps({
  id: {
    type: String,
    required: false,
    default: "",
  },
  colorAttr: {
    type: Object,
    required: true,
  },
  buttonVariant: {
    type: String as PropType<ButtonVariant>,
    required: true,
  },
});

const router = useRouter();
const memoForm = ref();
const memoStore = memo();
const foundMemo = memoStore.findById(props.id.valueOf());

const titleInput = ref(foundMemo && foundMemo.title ? foundMemo.title : "");
const contentInput = ref(
  foundMemo && foundMemo.content ? foundMemo.content : ""
);

/**
 * determine if this component is used for edit page
 *
 * @returns {boolean}
 */
function isEdit(): boolean {
  return props.id.length > 0;
}

/**
 * stringify current date
 *
 * @returns {string}
 */
function stringifyCurrentDate(): string {
  const date = new Date();
  const year = date.getFullYear();
  const month = date.getMonth();
  const day = date.getDate();
  return year + month + day + Date.now() + "";
}

/**
 * create Memo type data
 *
 * @returns {Memo}
 */
function formMemo(): Memo {
  return {
    id: stringifyCurrentDate(),
    title: titleInput.value,
    content: contentInput.value,
    dateTime: moment().format("YYYY/MM/DD HH:mm"),
  };
}

/**
 * save to store
 *
 * @returns {Promise<void>}
 */
async function save(): Promise<void> {
  const validResult = await memoForm.value.validate();
  if (!validResult.valid) {
    return;
  }
  memoStore.add(formMemo());
  forwardHome();
}

/**
 * update to store data
 *
 * @returns {Promise<void>}
 */
async function update(): Promise<void> {
  const validResult = await memoForm.value.validate();
  if (!validResult.valid) {
    return;
  }
  memoStore.updateById(props.id, titleInput.value, contentInput.value);
  forwardHome();
}

/**
 * remove from store data
 *
 * @returns {void}
 */
function remove(): void {
  memoStore.removeById(props.id);
  forwardHome();
}

/**
 * forward home page
 *
 * @returns {void}
 */
function forwardHome(): void {
  router.push("/");
}
</script>
