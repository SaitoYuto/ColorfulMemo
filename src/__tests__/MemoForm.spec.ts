import MemoForm from "@/components/MemoForm.vue";
import RequiredTextField from "@/components/RequiredTextField.vue";
import RequiredTextArea from "@/components/RequiredTextArea.vue";
import { BUTTON_VARIANT } from "@/constants/ButtonVariant";
import { COLOR_ATTR } from "@/constants/ColorAttr";
import { useMemoStore } from "@/stores/memo";
import { flushPromises, mount } from "@vue/test-utils";
import "@/__tests__/setup";
import { mockRouter } from "@/__tests__/setup";
import moment from "moment";
import { MockInstance } from "vitest";
import { INTERFACE_LABEL } from "@/constants/InterfaceLabel";

describe("MemoForm.vue", () => {
  let memoStore: ReturnType<typeof useMemoStore>;
  let routerPushSpy: MockInstance;

  beforeEach(() => {
    memoStore = useMemoStore();
    vi.mock("vue-router", () => ({
      useRouter: () => mockRouter,
    }));
    routerPushSpy = vi.spyOn(mockRouter, "push");
  });

  afterEach(() => {
    vi.restoreAllMocks();
  });

  it("renders form for adding memo when not passing props.id", async () => {
    const emptyId = "";
    const wrapper = mount(MemoForm, {
      props: {
        id: emptyId,
        colorAttr: COLOR_ATTR.DEFAULT,
        buttonVariant: BUTTON_VARIANT.TONAL,
      },
    });

    const saveButton = await wrapper.findComponent("[data-testid=save-button]");
    const updateButton = await wrapper.findComponent(
      "[data-testid=update-button]"
    );
    const deleteButton = await wrapper.findComponent(
      "[data-testid=delete-button]"
    );
    const cancelButton = await wrapper.findComponent(
      "[data-testid=cancel-button]"
    );
    expect(saveButton.isVisible()).toBe(true);
    expect(saveButton.text()).toBe(INTERFACE_LABEL.BUTTON.SAVE);
    expect(updateButton.isVisible()).toBe(false);
    expect(updateButton.text()).toBe(INTERFACE_LABEL.BUTTON.UPDATE);
    expect(deleteButton.isVisible()).toBe(false);
    expect(deleteButton.text()).toBe(INTERFACE_LABEL.BUTTON.DELETE);
    expect(cancelButton.isVisible()).toBe(true);
    expect(cancelButton.text()).toBe(INTERFACE_LABEL.BUTTON.CANCEL);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders form for editing existing memo when passing props.id", async () => {
    const id = "123";
    const wrapper = mount(MemoForm, {
      props: {
        id: id,
        colorAttr: COLOR_ATTR.DEFAULT,
        buttonVariant: BUTTON_VARIANT.TONAL,
      },
    });
    const saveButton = await wrapper.findComponent("[data-testid=save-button]");
    const updateButton = await wrapper.findComponent(
      "[data-testid=update-button]"
    );
    const deleteButton = await wrapper.findComponent(
      "[data-testid=delete-button]"
    );
    const cancelButton = await wrapper.findComponent(
      "[data-testid=cancel-button]"
    );
    expect(saveButton.isVisible()).toBe(false);
    expect(saveButton.text()).toBe(INTERFACE_LABEL.BUTTON.SAVE);
    expect(updateButton.isVisible()).toBe(true);
    expect(updateButton.text()).toBe(INTERFACE_LABEL.BUTTON.UPDATE);
    expect(deleteButton.isVisible()).toBe(true);
    expect(deleteButton.text()).toBe(INTERFACE_LABEL.BUTTON.DELETE);
    expect(cancelButton.isVisible()).toBe(true);
    expect(cancelButton.text()).toBe(INTERFACE_LABEL.BUTTON.CANCEL);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("saves new memo correctly", async () => {
    const addSpy = vi.spyOn(memoStore, "add");
    const emptyId = "";
    const wrapper = mount(MemoForm, {
      props: {
        id: emptyId,
        colorAttr: COLOR_ATTR.DEFAULT,
        buttonVariant: BUTTON_VARIANT.TONAL,
      },
    });
    const titleValue = "New Memo Title";
    const contentValue = "New Memo Content";
    await wrapper
      .findComponent<typeof RequiredTextField>("[data-testid=title-text-field]")
      .vm.$emit("update:input", titleValue);
    await wrapper
      .findComponent<typeof RequiredTextArea>("[data-testid=content-text-area]")
      .vm.$emit("update:input", contentValue);
    await wrapper.findComponent("[data-testid=save-button]").trigger("click");
    await flushPromises();
    expect(addSpy).toHaveBeenCalledTimes(1);
    expect(addSpy).toHaveBeenCalledWith(
      expect.objectContaining({
        title: titleValue,
        content: contentValue,
      })
    );
    expect(routerPushSpy).toHaveBeenCalledWith("/");
  });

  it("fails to save new memo for validate violation ", async () => {
    const addSpy = vi.spyOn(memoStore, "add");
    const emptyId = "";
    const wrapper = mount(MemoForm, {
      props: {
        id: emptyId,
        colorAttr: COLOR_ATTR.DEFAULT,
        buttonVariant: BUTTON_VARIANT.TONAL,
      },
    });
    const titleValue = "";
    const contentValue = "";
    await wrapper
      .findComponent<typeof RequiredTextField>("[data-testid=title-text-field]")
      .vm.$emit("update:input", titleValue);
    await wrapper
      .findComponent<typeof RequiredTextArea>("[data-testid=content-text-area]")
      .vm.$emit("update:input", contentValue);
    await wrapper.findComponent("[data-testid=save-button]").trigger("click");
    await flushPromises();
    expect(addSpy).toHaveBeenCalledTimes(0);
    expect(addSpy).not.toHaveBeenCalled();
    expect(routerPushSpy).not.toHaveBeenCalledWith("/");
  });

  it("updates existing memo correctly", async () => {
    const updateSpy = vi.spyOn(memoStore, "updateById");
    const id = "123";
    const wrapper = mount(MemoForm, {
      props: {
        id: id,
        colorAttr: COLOR_ATTR.DEFAULT,
        buttonVariant: BUTTON_VARIANT.TONAL,
      },
    });

    memoStore.add({
      id: id,
      title: "Old Memo Title",
      content: "Old Memo Content",
      dateTime: moment().format("YYYY/MM/DD HH:mm"),
    });
    const titleValue = "New Memo Title";
    const contentValue = "New Memo Content";
    await wrapper
      .findComponent<typeof RequiredTextField>("[data-testid=title-text-field]")
      .vm.$emit("update:input", titleValue);
    await wrapper
      .findComponent<typeof RequiredTextArea>("[data-testid=content-text-area]")
      .vm.$emit("update:input", contentValue);
    await wrapper.findComponent("[data-testid=update-button]").trigger("click");
    await flushPromises();
    expect(updateSpy).toHaveBeenCalledTimes(1);
    expect(updateSpy).toHaveBeenCalledWith(id, titleValue, contentValue);
    expect(routerPushSpy).toHaveBeenCalledWith("/");
  });

  it("fails to update existing memo for validate violation", async () => {
    const updateSpy = vi.spyOn(memoStore, "updateById");
    const id = "123";
    const wrapper = mount(MemoForm, {
      props: {
        id: id,
        colorAttr: COLOR_ATTR.DEFAULT,
        buttonVariant: BUTTON_VARIANT.TONAL,
      },
    });

    memoStore.add({
      id: id,
      title: "Old Memo Title",
      content: "Old Memo Content",
      dateTime: moment().format("YYYY/MM/DD HH:mm"),
    });
    const titleValue = "";
    const contentValue = "";
    await wrapper
      .findComponent<typeof RequiredTextField>("[data-testid=title-text-field]")
      .vm.$emit("update:input", titleValue);
    await wrapper
      .findComponent<typeof RequiredTextArea>("[data-testid=content-text-area]")
      .vm.$emit("update:input", contentValue);
    await wrapper.findComponent("[data-testid=update-button]").trigger("click");
    await flushPromises();
    expect(updateSpy).toHaveBeenCalledTimes(0);
    expect(updateSpy).not.toHaveBeenCalledWith(id, titleValue, contentValue);
    expect(routerPushSpy).not.toHaveBeenCalledWith("/");
  });

  it("removes existing memo correctly", async () => {
    const removeSpy = vi.spyOn(memoStore, "removeById");
    const id = "123";
    const wrapper = mount(MemoForm, {
      props: {
        id: id,
        colorAttr: COLOR_ATTR.DEFAULT,
        buttonVariant: BUTTON_VARIANT.TONAL,
      },
    });

    memoStore.add({
      id: id,
      title: "Memo Title",
      content: "Memo Content",
      dateTime: moment().format("YYYY/MM/DD HH:mm"),
    });

    await wrapper.findComponent("[data-testid=delete-button]").trigger("click");
    await flushPromises();
    expect(removeSpy).toHaveBeenCalledTimes(1);
    expect(removeSpy).toHaveBeenCalledWith(id);
    expect(routerPushSpy).toHaveBeenCalledWith("/");
  });

  it("clicks cancel button on form for adding memo", async () => {
    const emptyId = "";
    const wrapper = mount(MemoForm, {
      props: {
        id: emptyId,
        colorAttr: COLOR_ATTR.DEFAULT,
        buttonVariant: BUTTON_VARIANT.TONAL,
      },
    });

    await wrapper.findComponent("[data-testid=cancel-button]").trigger("click");
    expect(routerPushSpy).toHaveBeenCalledWith("/");
  });

  it("clicks cancel button on form for editing memo", async () => {
    const id = "123";
    const wrapper = mount(MemoForm, {
      props: {
        id: id,
        colorAttr: COLOR_ATTR.DEFAULT,
        buttonVariant: BUTTON_VARIANT.TONAL,
      },
    });

    await wrapper.findComponent("[data-testid=cancel-button]").trigger("click");
    expect(routerPushSpy).toHaveBeenCalledWith("/");
  });
});
