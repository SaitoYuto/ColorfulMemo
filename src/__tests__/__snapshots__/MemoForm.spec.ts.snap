// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MemoForm.vue > renders form for adding memo when not passing props.id 1`] = `
"<form class="v-form" novalidate="">
  <div class="v-container v-locale--is-ltr">
    <div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-text-field mt-4" data-testid="title-text-field">
      <!---->
      <div class="v-input__control">
        <div class="v-field v-field--appended v-field--center-affix v-field--has-background v-field--variant-outlined v-theme--light v-locale--is-ltr" style="background-color: rgb(245, 245, 245); color: rgb(0, 0, 0); caret-color: #000;">
          <div class="v-field__overlay"></div>
          <div class="v-field__loader">
            <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
              <!---->
              <div class="v-progress-linear__background" style="width: 100%;"></div>
              <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                <div class="v-progress-linear__indeterminate">
                  <div class="v-progress-linear__indeterminate long"></div>
                  <div class="v-progress-linear__indeterminate short"></div>
                </div>
              </transition-stub>
              <!---->
            </div>
          </div>
          <!---->
          <div class="v-field__field" data-no-activator="">
            <!----><label class="v-label v-field-label" for="input-0">
              <!---->*Title
            </label>
            <!----><input size="1" type="text" id="input-0" aria-describedby="input-0-messages" rows="1" class="v-field__input">
            <!---->
          </div>
          <transition-stub name="expand-x-transition" appear="false" persisted="false" css="true">
            <div class="v-field__clearable" style="display: none;"><i class="mdi-close-circle mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" aria-hidden="false" tabindex="0" aria-label="Clear *Title"></i></div>
          </transition-stub>
          <div class="v-field__append-inner">
            <!---->
            <!---->
          </div>
          <div class="v-field__outline">
            <div class="v-field__outline__start"></div>
            <div class="v-field__outline__notch"><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-0">
                <!---->*Title
              </label></div>
            <div class="v-field__outline__end"></div>
            <!---->
          </div>
        </div>
      </div>
      <!---->
      <div class="v-input__details">
        <transition-group-stub name="slide-y-transition" tag="div" appear="false" persisted="false" css="true" class="v-messages" role="alert" aria-live="polite" id="input-0-messages">
          <!---->
        </transition-group-stub>
        <!---->
      </div>
    </div>
    <div class="v-input v-input--horizontal v-input--density-default v-locale--is-ltr v-textarea v-text-field mt-4" data-testid="content-text-area">
      <!---->
      <div class="v-input__control">
        <div class="v-field v-field--appended v-field--has-background v-field--variant-outlined v-theme--light v-locale--is-ltr" style="background-color: rgb(245, 245, 245); color: rgb(0, 0, 0); caret-color: #000;">
          <div class="v-field__overlay"></div>
          <div class="v-field__loader">
            <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
              <!---->
              <div class="v-progress-linear__background" style="width: 100%;"></div>
              <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                <div class="v-progress-linear__indeterminate">
                  <div class="v-progress-linear__indeterminate long"></div>
                  <div class="v-progress-linear__indeterminate short"></div>
                </div>
              </transition-stub>
              <!---->
            </div>
          </div>
          <!---->
          <div class="v-field__field" data-no-activator="">
            <!----><label class="v-label v-field-label" for="input-2">
              <!---->*Content
            </label>
            <!----><textarea class="v-field__input" rows="5" id="input-2" aria-describedby="input-2-messages"></textarea>
            <!---->
            <!---->
          </div>
          <transition-stub name="expand-x-transition" appear="false" persisted="false" css="true">
            <div class="v-field__clearable" style="display: none;"><i class="mdi-close-circle mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" aria-hidden="false" tabindex="0" aria-label="Clear *Content"></i></div>
          </transition-stub>
          <div class="v-field__append-inner">
            <!---->
            <!---->
          </div>
          <div class="v-field__outline">
            <div class="v-field__outline__start"></div>
            <div class="v-field__outline__notch"><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-2">
                <!---->*Content
              </label></div>
            <div class="v-field__outline__end"></div>
            <!---->
          </div>
        </div>
      </div>
      <!---->
      <div class="v-input__details">
        <transition-group-stub name="slide-y-transition" tag="div" appear="false" persisted="false" css="true" class="v-messages" role="alert" aria-live="polite" id="input-2-messages">
          <!---->
        </transition-group-stub>
        <!---->
      </div>
    </div>
  </div>
  <div class="v-container v-locale--is-ltr d-flex justify-space-evenly mx-16"><button type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-tonal" style="color: rgb(158, 158, 158); caret-color: #9E9E9E;" data-testid="save-button"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator="">Save</span>
      <!---->
      <!---->
    </button><button type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-tonal" style="color: rgb(158, 158, 158); caret-color: #9E9E9E; display: none;" data-testid="update-button"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator="">Update</span>
      <!---->
      <!---->
    </button><button type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-tonal" style="color: rgb(158, 158, 158); caret-color: #9E9E9E; display: none;" data-testid="delete-button"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator="">Delete</span>
      <!---->
      <!---->
    </button><button type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-tonal" style="color: rgb(158, 158, 158); caret-color: #9E9E9E;" data-testid="cancel-button"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator="">Cancel</span>
      <!---->
      <!---->
    </button></div>
</form>"
`;

exports[`MemoForm.vue > renders form for editing existing memo when passing props.id 1`] = `
"<form class="v-form" novalidate="">
  <div class="v-container v-locale--is-ltr">
    <div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-text-field mt-4" data-testid="title-text-field">
      <!---->
      <div class="v-input__control">
        <div class="v-field v-field--appended v-field--center-affix v-field--has-background v-field--variant-outlined v-theme--light v-locale--is-ltr" style="background-color: rgb(245, 245, 245); color: rgb(0, 0, 0); caret-color: #000;">
          <div class="v-field__overlay"></div>
          <div class="v-field__loader">
            <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
              <!---->
              <div class="v-progress-linear__background" style="width: 100%;"></div>
              <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                <div class="v-progress-linear__indeterminate">
                  <div class="v-progress-linear__indeterminate long"></div>
                  <div class="v-progress-linear__indeterminate short"></div>
                </div>
              </transition-stub>
              <!---->
            </div>
          </div>
          <!---->
          <div class="v-field__field" data-no-activator="">
            <!----><label class="v-label v-field-label" for="input-0">
              <!---->*Title
            </label>
            <!----><input size="1" type="text" id="input-0" aria-describedby="input-0-messages" rows="1" class="v-field__input">
            <!---->
          </div>
          <transition-stub name="expand-x-transition" appear="false" persisted="false" css="true">
            <div class="v-field__clearable" style="display: none;"><i class="mdi-close-circle mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" aria-hidden="false" tabindex="0" aria-label="Clear *Title"></i></div>
          </transition-stub>
          <div class="v-field__append-inner">
            <!---->
            <!---->
          </div>
          <div class="v-field__outline">
            <div class="v-field__outline__start"></div>
            <div class="v-field__outline__notch"><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-0">
                <!---->*Title
              </label></div>
            <div class="v-field__outline__end"></div>
            <!---->
          </div>
        </div>
      </div>
      <!---->
      <div class="v-input__details">
        <transition-group-stub name="slide-y-transition" tag="div" appear="false" persisted="false" css="true" class="v-messages" role="alert" aria-live="polite" id="input-0-messages">
          <!---->
        </transition-group-stub>
        <!---->
      </div>
    </div>
    <div class="v-input v-input--horizontal v-input--density-default v-locale--is-ltr v-textarea v-text-field mt-4" data-testid="content-text-area">
      <!---->
      <div class="v-input__control">
        <div class="v-field v-field--appended v-field--has-background v-field--variant-outlined v-theme--light v-locale--is-ltr" style="background-color: rgb(245, 245, 245); color: rgb(0, 0, 0); caret-color: #000;">
          <div class="v-field__overlay"></div>
          <div class="v-field__loader">
            <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
              <!---->
              <div class="v-progress-linear__background" style="width: 100%;"></div>
              <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                <div class="v-progress-linear__indeterminate">
                  <div class="v-progress-linear__indeterminate long"></div>
                  <div class="v-progress-linear__indeterminate short"></div>
                </div>
              </transition-stub>
              <!---->
            </div>
          </div>
          <!---->
          <div class="v-field__field" data-no-activator="">
            <!----><label class="v-label v-field-label" for="input-2">
              <!---->*Content
            </label>
            <!----><textarea class="v-field__input" rows="5" id="input-2" aria-describedby="input-2-messages"></textarea>
            <!---->
            <!---->
          </div>
          <transition-stub name="expand-x-transition" appear="false" persisted="false" css="true">
            <div class="v-field__clearable" style="display: none;"><i class="mdi-close-circle mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" aria-hidden="false" tabindex="0" aria-label="Clear *Content"></i></div>
          </transition-stub>
          <div class="v-field__append-inner">
            <!---->
            <!---->
          </div>
          <div class="v-field__outline">
            <div class="v-field__outline__start"></div>
            <div class="v-field__outline__notch"><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-2">
                <!---->*Content
              </label></div>
            <div class="v-field__outline__end"></div>
            <!---->
          </div>
        </div>
      </div>
      <!---->
      <div class="v-input__details">
        <transition-group-stub name="slide-y-transition" tag="div" appear="false" persisted="false" css="true" class="v-messages" role="alert" aria-live="polite" id="input-2-messages">
          <!---->
        </transition-group-stub>
        <!---->
      </div>
    </div>
  </div>
  <div class="v-container v-locale--is-ltr d-flex justify-space-evenly mx-16"><button type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-tonal" style="color: rgb(158, 158, 158); caret-color: #9E9E9E; display: none;" data-testid="save-button"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator="">Save</span>
      <!---->
      <!---->
    </button><button type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-tonal" style="color: rgb(158, 158, 158); caret-color: #9E9E9E;" data-testid="update-button"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator="">Update</span>
      <!---->
      <!---->
    </button><button type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-tonal" style="color: rgb(158, 158, 158); caret-color: #9E9E9E;" data-testid="delete-button"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator="">Delete</span>
      <!---->
      <!---->
    </button><button type="button" class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-tonal" style="color: rgb(158, 158, 158); caret-color: #9E9E9E;" data-testid="cancel-button"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator="">Cancel</span>
      <!---->
      <!---->
    </button></div>
</form>"
`;
